### PR TITLE
Add more items in apple targets

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
@JohnTitor Is it okay to upgrade the minimum supported rust version? Otherwise I'll simply remove `vnode_info_path` and `proc_vnodepathinfo` from the added structs.

File: src/unix/bsd/apple/mod.rs [hunk 1]
```diff
@@ -678,6 +678,47 @@ s! {
...
+    pub struct vnode_info {
+        pub vi_stat: vinfo_stat,
+        pub vi_type: ::c_int,
+        pub vi_pad: ::c_int,
+        pub vi_fsid: ::fsid_t,
+    }
+
+    pub struct vnode_info_path {
+        pub vip_vi: vnode_info,
+        pub vip_path: [::c_char; ::MAXPATHLEN as usize],
```
Comment (JohnTitor) [inline src/unix/bsd/apple/mod.rs:715]: We could nest the array like `[[::c_char; 32]; 32]` to auto-derive traits.

Comment (GuillaumeGomez) [reply src/unix/bsd/apple/mod.rs:715]: Wow, quite the dark magic, I like it!

**Comment by GuillaumeGomez** [comment]
@JohnTitor: The checkers aren't happy, so I guess this solution cannot work...

**Comment by GuillaumeGomez** [comment]
@JohnTitor Seems like it worked! \o/

**Comment by JohnTitor** [comment]
Thanks!
@bors r+

**Pull request merged**